src/test/java/uk/gov/hmcts/reform/sscs/factory/NotificationFactoryTest.java

```diff
@@ -248,41 +253,47 @@ public void buildLastNotificationFromSscsCaseDataEventWhenSmsFirstSubscribed() {
                 .notificationEventType(SUBSCRIPTION_UPDATED_NOTIFICATION)
                 .build();
 
-        Notification result = factory.create(new CcdNotificationWrapper(wrapper), APPELLANT);
+        Notification result = factory.create(new CcdNotificationWrapper(wrapper), getSubscriptionWithType(wrapper, APPELLANT));
 
+        assertNull(result.getDestination().email);
+        assertNotNull(subscription.getMobile());
+        assertEquals(subscription.getMobile(), result.getDestination().sms);
         assertEquals("123", result.getSmsTemplate());
     }
 
     @Test
-    public void buildNoNotificationFromSscsCaseDataWhenSubscriptionUpdateReceivedWithNoChangeInEmailAddress() {
+    public void buildNoNotificationFromSscsCaseDataWhenSubscriptionUpdateReceivedWithNoChangeInSubscription() {
         when(personalisationFactory.apply(SUBSCRIPTION_UPDATED_NOTIFICATION)).thenReturn(subscriptionPersonalisation);
-        when(config.getTemplate(DO_NOT_SEND.getId(), SUBSCRIPTION_CREATED_NOTIFICATION.getId(), DO_NOT_SEND.getId(), PIP, REGULAR)).thenReturn(Template.builder().emailTemplateId(null).smsTemplateId(null).build());
+        when(config.getTemplate(SUBSCRIPTION_UPDATED_NOTIFICATION.getId(), SUBSCRIPTION_UPDATED_NOTIFICATION.getId(), SUBSCRIPTION_UPDATED_NOTIFICATION.getId(), PIP, REGULAR)).thenReturn(Template.builder().emailTemplateId("123").smsTemplateId("123").build());
 
         List<Event> events = new ArrayList<>();
         events.add(Event.builder().value(EventDetails.builder().date(DATE).type(APPEAL_RECEIVED_NOTIFICATION.getId()).build()).build());
 
-        wrapper = SscsCaseDataWrapper.builder()
+        SscsCaseDataWrapper wrapper = SscsCaseDataWrapper.builder()
                 .newSscsCaseData(
                         ccdResponse.toBuilder()
                                 .subscriptions(Subscriptions.builder().appellantSubscription(subscription.toBuilder().subscribeSms("Yes").subscribeEmail("Yes").build()).build())
                                 .events(events)
                                 .build())
                 .oldSscsCaseData(
                         ccdResponse.toBuilder()
-                                .subscriptions(Subscriptions.builder().appellantSubscription(subscription.toBuilder().subscribeSms("No").subscribeEmail("Yes").build()).build())
+                                .subscriptions(Subscriptions.builder().appellantSubscription(subscription.toBuilder().subscribeSms("Yes").subscribeEmail("Yes").build()).build())
                                 .build())
                 .notificationEventType(SUBSCRIPTION_UPDATED_NOTIFICATION)
                 .build();
 
-        Notification result = factory.create(new CcdNotificationWrapper(wrapper), APPELLANT);
+        Notification result = factory.create(new CcdNotificationWrapper(wrapper), getSubscriptionWithType(wrapper, APPELLANT));
 
-        assertNull(result.getEmailTemplate());
+        assertEquals("123", result.getEmailTemplate());
```

Comment:
This doesn't look correct to me? The test name doesn't want a notification to be sent but you have changed the assertions to check that an email and sms template exist.

Reply:
It used to change the notification event to `donotsend` and hence has a null email and sms template.
Now, It doesn't get sent due to the email and sms being null 2 lines below.